### PR TITLE
Adding missing browser versions

### LINKED ISSUE #1664: Missing browser/app versions for October
Trying to make this a monthly thing to keep up on browser releases.

### PR DESCRIPTION
**Pull request by jaydiablo**
Fixes #1664

I looked for all UAs that parsed with a version of 0 from October (I had just looked at the major version for this, which is why some UAs with a major version of “0” and a minor version of non-“0” snuck into the tests. I think that’s fine though, I did look at all of the UAs in the tests to be sure they’re parsed properly).

I added versions where necessary and in a few cases I added some new platforms (mainly so that the UAs were parsed well for the tests).

I also noticed that Firefox on Android had a version of Android 6.1 in the platforms, which was never an Android release (we removed other versions of this platform in the past, but must’ve missed this particular style of pattern).

Most of the UAs I pulled now parse with a proper version number, but I only added the ones to the tests that were needed for coverage, and that didn’t have the “general Mobile Phone/Tablet/Device” devices.

### CODE REVIEW

File: resources/user-agents/apps/apple/apple-mail.json [hunk 1]
```diff
@@ -247,6 +247,18 @@
         {
           "match": "Mozilla/5.0 (#PLATFORM#) applewebkit* (*khtml*like*gecko*)",
           "properties": {
+            "Version": "11.0",
```
Comment (jaydiablo) [inline resources/user-agents/apps/apple/apple-mail.json]: I confirmed this version on macOS 10.13

File: resources/user-agents/apps/apple/apple-mail.json [hunk 2]
```diff
@@ -373,24 +385,44 @@
         },
         {
           "match": "Mail/* CFNetwork/760*",
+          "properties": {
```
Comment (jaydiablo) [inline resources/user-agents/apps/apple/apple-mail.json]: I made these match the Mozilla patterns above, which required a couple of changes to some tests.

File: resources/user-agents/browsers/firefox-on-android/firefox-android-32-to-45.json [hunk 3]
```diff
@@ -33,7 +33,7 @@
             "AndroidMobile_2_3",
```
Comment (mimmi20) [inline resources/user-agents/browsers/firefox-on-android/firefox-android-32-to-45.json]: please flip the sorting the platforms here, like did in the `firefox-android-48-on.json` file

File: resources/user-agents/browsers/firefox-on-android/firefox-android-46-to-47.json [hunk 4]
```diff
@@ -33,7 +33,7 @@
             "AndroidMobile_2_3",
```
Comment (mimmi20) [inline resources/user-agents/browsers/firefox-on-android/firefox-android-46-to-47.json]: please flip the sorting the platforms here, like did in the `firefox-android-48-on.json` file

File: resources/user-agents/browsers/chrome/chrome-ios-48-on.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "division": "Chrome #MAJORVER#.#MINORVER# for iOS",
-  "versions": [63, 62, 61, 60, 59, 58, 57, 56, 55, 54, 53, 52, 51, 50, 49, 48],
+  "versions": [64, 63, 62, 61, 60, 59, 58, 57, 56, 55, 54, 53, 52, 51, 50, 49, 48],
```
Comment (mimmi20) [inline resources/user-agents/browsers/chrome/chrome-ios-48-on.json]: Why did you not add iOS 11.1 here too?

Comment (jaydiablo) [reply resources/user-agents/browsers/chrome/chrome-ios-48-on.json]: I have it coming in the PR I have prepped for #1666

**Comment by jaydiablo** [comment]
The reduction in coverage reported here is because the last merge to master only reported the PHP coverage results, not the JSON files' coverage.

Coverage looks as expected though: https://codecov.io/gh/browscap/browscap/tree/0a0e04d932298090f6ddf2bbbfae26bc6328131c

File: resources/user-agents/browsers/amigo/amigo-32-to-56.json [hunk 6]
```diff
@@ -1,6 +1,6 @@
 {
   "division": "Amigo #MAJORVER#.#MINORVER#",
-  "versions": [56, 55, 54, 53, 52, 51, 50, 47, 46, 45, 44, 43, 42, 36, 35, 34, 33, 32],
+  "versions": [58, 57, 56, 55, 54, 53, 52, 51, 50, 47, 46, 45, 44, 43, 42, 36, 35, 34, 33, 32],
```
[the remark this answers is not in the document]
Comment (jaydiablo) [reply resources/user-agents/browsers/amigo/amigo-32-to-56.json]: Changed to `amigo-32-on.json`

File: resources/user-agents/browsers/firefox-on-android/firefox-android-32-to-45.json (hunk 3 above)
Comment (jaydiablo) [reply resources/user-agents/browsers/firefox-on-android/firefox-android-32-to-45.json]: flipped

File: resources/user-agents/browsers/firefox-on-android/firefox-android-46-to-47.json (hunk 4 above)
Comment (jaydiablo) [reply resources/user-agents/browsers/firefox-on-android/firefox-android-46-to-47.json]: flipped

File: resources/user-agents/browsers/chrome/chrome-ios-48-on.json (hunk 5 above)
Comment (mimmi20) [reply resources/user-agents/browsers/chrome/chrome-ios-48-on.json]: ok

**Pull request merged**